File: Snackdown2k19/Round1A/weirdsum.cpp
#include<iostream>
#include<string>
#include<bits/stdc++.h>
#include <map>
#include <vector>
#include <unordered_set>
#include <iterator>

using namespace std;

int main(int argc, char const *argv[])
{
	int t;
	cin>>t;

	while(t--)
	{
		long long int a,b,c;
		cin>>a;
		cin>>b;
		cin>>c;

		long long int a_one = __builtin_popcountll(a);
		long long int b_one = __builtin_popcountll(b);

		long long int count=0;
		
		long long int ar[c];
		memset(ar, 0, sizeof(ar));

		for (long long int i = 0; i <=c; ++i)
		{
			if(ar[i]==0)
			{
				ar[i] = __builtin_popcountll(i);
				ar[c-i] = __builtin_popcountll(c-i);
			}

			if(ar[i]==a_one && ar[c-i]==b_one)
			{
				count++;
			}
		}

		cout<<count<<endl;


	}
	return 0;
}

File: Snackdown2k19/Round1A/weirdsum.py
t = int(input())
for i in range(t):
	
	a,b,c=input().split(" ")

	a=int(a)
	a_one = bin(a).count("1")
	#print(bin(a),a_one)

	b=int(b)
	b_one = bin(b).count("1")
	#print(b_one)

	c=int(c)
	count=0
	binary=[0]*(c+1)
	#print(binary)
	for i in range(c+1):
		if(binary[i]==0):
			binary[i]=bin(i).count("1")

		if(binary[c-i]==0):
			binary[c-i]=bin(c-i).count("1")

		if(binary[i]==a_one and binary[c-i]==b_one):
			count+=1

	print(count)
	
		